src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateResults = [];
+  let index = 0;
+  let copyState;
+
+  for (const action of actions) {
+    if (index === 0) {
+      copyState = { ...state };
+    } else {
+      copyState = { ...stateResults[index - 1] };
+    }
+    index++;
```

Comment:
it's much easier to use some temp variable for example 'tempState' and operate with it. But then at the end of `for loop` you can add to the stateResults a copy of the tempState